lib/matplotlib/font_manager.py

```diff
@@ -180,24 +184,46 @@ def win32InstalledFonts(directory=None, fontext='ttf'):
 
     items = set()
     for fontdir in MSFontDirectories:
+        # System fonts
```

Comment:
The logic in this is complicated enough, can you put it in a helper function that we call twice?

Reply:
although, I now see there is some subtlety with absolute vs relative paths.